components/dashboard/src/app/AppLoading.tsx

```diff
@@ -5,8 +5,19 @@
  */
 
 import { FunctionComponent } from "react";
+import { Delayed } from "../components/Delayed";
+import { Heading3, Subheading } from "../components/typography/headings";
+import gitpodIcon from "../icons/gitpod.svg";
 
-// TODO: Would be great to show a loading UI if it's been loading for awhile
 export const AppLoading: FunctionComponent = () => {
-    return <></>;
+    return (
+        // Wait 3 seconds before showing the loading screen to avoid flashing it too quickly
+        <Delayed wait={3000}>
```

Comment:
**question**: 3 seconds sound a bit too much, no? Could it be better to make this 1 second?

Reply:
Lets try 2. I want to avoid it flashing too quickly, 1 second is pretty easy to bump into even w/ a good connection if there's 1 slow call.